toolchain/sem_ir/typed_insts.h

```diff
@@ -795,6 +795,29 @@ struct FunctionTypeWithSelfType {
   InstId self_id;
 };
 
+// An overloaded C++ function declaration.
+struct OverloadedCppFunctionDecl {
+  static constexpr auto Kind =
+      InstKind::OverloadedCppFunctionDecl.Define<Parse::AnyFunctionDeclId>(
+          {.ir_name = "overloaded_cpp_fn_decl", .is_lowered = false});
```

Comment:
This isn't really a declaration; how about calling it `OverloadedCppFunctionValue` instead?
```suggestion
// An unresolved overloaded C++ function value.
struct OverloadedCppFunctionValue {
  static constexpr auto Kind =
      InstKind::OverloadedCppFunctionValue.Define<Parse::NodeId>(
          {.ir_name = "overloaded_cpp_fn_value", .is_lowered = false});
```

Reply:
Done. (I changed the name to `CppOverloadSetValue` as per the comment above).